### ISSUE TITLE
scikit-learn 1.5.0 breaks glum

### ISSUE THREAD

**Issue by jtilly** [Quantco/glum#793]
```
Python 3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:35:20) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import glum
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/tmp/glum/lib/python3.12/site-packages/glum/__init__.py", line 14, in <module>
    from ._glm import GeneralizedLinearRegressor, get_family, get_link
  File "/private/tmp/glum/lib/python3.12/site-packages/glum/_glm.py", line 58, in <module>
    from ._solvers import (
  File "/private/tmp/glum/lib/python3.12/site-packages/glum/_solvers.py", line 13, in <module>
    from ._cd_fast import (
  File "src/glum/_cd_fast.pyx", line 1, in init glum._cd_fast
TypeError: C variable sklearn.utils._random.DEFAULT_SEED has wrong signature (expected __pyx_t_7sklearn_5utils_7_random_UINT32_t, got __pyx_t_7sklearn_5utils_9_typedefs_uint32_t)
>>> 
```

This happens with
```
glum                      3.0.0           py312hed2d4af_0    conda-forge
scikit-learn              1.5.0           py312h1b546db_0    conda-forge
```

The relevant change happened in https://github.com/scikit-learn/scikit-learn/pull/28656.

https://github.com/scikit-learn/scikit-learn/pull/28656/files#diff-5829233c37ebf7320b24dd970725c53fbbb1cd789e2ad5157093280f7f2a388cR8

I guess our CI system didn't catch this, because there we're always testing against the version that we're building with.

**Comment by MatthiasSchmidtblaicherQC** [comment]
Should be fixed by [this change](https://github.com/Quantco/glum/pull/794).

**Comment by MatthiasSchmidtblaicherQC** [comment]
To fix this without requiring ``sklearn >= 1.5``, we will need to implement the random integer generation. Meanwhile, ``sklearn<1.5`` has been pinned repodata for past versions [here](https://github.com/conda-forge/glum-feedstock/pull/32) and [here](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/734).
